pennylane_lightning/src/gates/cpu_kernels/avx_common/AVXUtil.hpp

```diff
@@ -265,6 +212,53 @@ template <size_t packed_size> struct InternalWires {
 template <size_t packed_size>
 constexpr auto internal_wires_v = InternalWires<packed_size>::value;
 
+template <typename PrecisionT, size_t packed_size> struct Set;
+#ifdef PL_USE_AVX2
+template <> struct Set<float, 8> {
+    constexpr static auto create(const std::array<float, 8> &arr)
+        -> AVXIntrinsicType<float, 8> {
+        // NOLINTBEGIN(readability-magic-numbers)
+        return __m256{arr[0], arr[1], arr[2], arr[3],
+                      arr[4], arr[5], arr[6], arr[7]};
+        // NOLINTEND(readability-magic-numbers)
+    }
+};
```

Comment:
I wonder why you created several overloads for the `Set` struct where you only have a single method `create`.
After that, you call the function `set` (not to be confused with the `Set` struct), and this one only calls the `create` method. Would it be possible to only have the function `set` and overload it properly, as with the struct `Set`?

Reply:
Yeap, you are right. I just converted `set` to overloaded functions. Thanks for the nice catch!